Resources/js/router.js

```diff
@@ -1,187 +1,199 @@
-goog.provide('fos.Router');
-
-goog.require('goog.structs.Map');
-goog.require('goog.array');
-goog.require('goog.object');
-goog.require('goog.uri.utils');
+'use strict';
 
 /**
- * @constructor
- * @param {fos.Router.Context=} opt_context
- * @param {Object.<string, fos.Router.Route>=} opt_routes
+ * @fileoverview This file defines the Router class.
+ *
+ * You can compile this file by running the following command from the Resources folder:
+ *
+ *    npm install && npm run build
  */
-fos.Router = function(opt_context, opt_routes) {
-    this.context_ = opt_context || {base_url: '', prefix: '', host: '', scheme: ''};
-    this.setRoutes(opt_routes || {});
-};
-goog.addSingletonGetter(fos.Router);
 
 /**
- * @typedef {{
- *     tokens: (Array.<Array.<string>>),
- *     defaults: (Object.<string, string>),
- *     requirements: Object,
- *     hosttokens: (Array.<string>),
- *     schemes: (Array.<string>),
- *     methods: (Array.<string>)
- * }}
+ * Class Router
  */
-fos.Router.Route;
+class Router {
+
+    /**
+     * @constructor
+     * @param {Router.Context=} context
+     * @param {Object.<string, Router.Route>=} routes
+     */
+    constructor(context, routes) {
+        this.context_ = context || {base_url: '', prefix: '', host: '', scheme: ''};
+        this.setRoutes(routes || {});
+    }
 
-/**
- * @typedef {{
- *     base_url: (string)
- * }}
- */
-fos.Router.Context;
+    /**
+     * Returns the current instance.
+     * @returns {Router}
+     */
+    static getInstance() {
+        return Routing;
+    }
 
-/**
- * @param {Object.<string, fos.Router.Route>} routes
- */
-fos.Router.prototype.setRoutes = function(routes) {
-    this.routes_ = new goog.structs.Map(routes);
-};
+    /**
+     * Configures the current Router instance with the provided data.
+     * @param {Object} data
+     */
+    static setData(data) {
```

Comment:
I suggest to also add a non-static method to simplify the usage in a CommonJS environment (and then, the static method kept for BC can call the non-static method internally to avoid duplication)

Reply:
Good idea! now implemented.